File: app/src/main/kotlin/form/site/credleak.kt
package form.site

import io.ktor.server.netty.*
import io.ktor.server.routing.*
import io.ktor.server.application.*
import io.ktor.http.*
import io.ktor.server.response.*
import io.ktor.server.engine.*
import io.ktor.server.request.receiveParameters

import form.create.CredLeakFileBuilder
import form.create.createDocument
import form.events.Feed
import form.events.Event

fun Route.credLeak() {
    route("/credleak") {
        get {
            call.respondText(buildContent(createCredLeakPage(Event.buildEmpty()), Page.CredLeak), ContentType.Text.Html)
        }

        post {
            val params = call.receiveParameters().entries().associate { it.key.toString() to it.value.toString().removePrefix("[").removeSuffix("]")} as HashMap<String, String>
            val doc = CredLeakFileBuilder().buildFromMap(params)

            call.respondText("Created form: ${doc.getAlertName()}")
            createDocument(doc, params["ips"] == "on")
        }
    }
}

fun createCredLeakPage(e: Event): String {
    return """
    <form class="mt-4 flex flex-col justify-start" hx-post="/credleak" hx-target="#toast" hx-swap="innerHTML">
        <div class="flex">
            ${buildFormText("Name of organization", "orgName", default=e.name)}
            ${buildFormText("Form Number", "formNumber")}
        </div>
        ${buildFormText("Name of Victim Org", "victimOrg", default=e.name)}

        ${buildFormCheck("Multiple Leaks Found", "leakOrLeaks")}
        ${buildFormCheck("Multiple Credentials Leaked", "credOrCreds")}

        ${buildFormRadio(listOf("The passwords have been obfuscated to show only the first two letters.", "The passwords are not included due to being posted in plain text.", "The passwords are not included due to the threat actor not disclosing them.", "No Passwords were included"), "passwordOption", title="For passwords add either", checked=3)}

        ${buildFormText("Ip Address", "ipAddress", default=e.ip)}

        ${buildFormArea("Insert Username: Password", "userPass")}
        ${buildFormArea("Additional Information", "addInfo")}
        ${buildFormArea("Additional References", "additionalReferences")}

        ${buildFormRadio(listOf("Amber", "Green"), "changeTlp", title="TLP Alert" )}
        ${buildFormCheck("Create CSV", "ips")}

        <div class="flex justify-evenly">
            ${buildButton("Create", "submit")}
            <div class="min-w-2"></div>
            ${buildButton("Reset", "reset")}
        </div>
    </form>
    <div id="toast"></div>
    """
}

File: templates/credleak.go
package templates

import (

)

func CredLeak() string {
    data := struct {

    } {

    }

    const page = `
        <h1>Cred Leak</h1>
		<article>
			<form hx-post="/credleak" hx-target="body">
				<fieldset>
					<div class="grid">
						<label>
							Organization Name
							<input name="orgName"/>
						</label>
						<label>
							Form Number
							<input name="formNumber"/>
						</label>
					</div>
					<label>
						Victim Org Name
						<input name="victimOrg"/>
					</label>

					<label>
						<input type="checkbox" name="leaks"/>
						Multiple Leaks Found
					</label>

					<label>
						<input type="checkbox" name="creds"/>
						Multiple Creadentials
					</label>
					
					<hr>
					<label>For password add either:</label>
					<label>
						<input type="radio" value="The passwords have been obfuscated to show only the first two letters." name="password"/>
						The passwords have been obfuscated to show only the first two letters.  
					</label>
					<label>
						<input type="radio" value="The passwords are not included due to being posted in plain text." name="password"/>
						The passwords are not included due to being posted in plain text.
					</label>
					<label>
						<input type="radio" value="The passwords are not included due to the threat actor not disclosing them." name="password"/>
						The passwords are not included due to the threat actor not disclosing them.
					</label>
					<label>
						<input type="radio" value="No Passwords were included." name="password" checked/>
						No Passwords were included.
					</label>
					<hr>

					<label>
						IP Address
						<input name="ipAddress" />
					</label>
					
					<label>
						Insert Username: Password
						<textarea name="userPass"></textarea>
					</label>

					<label>
						Additional Information
						<textarea name="addInfo"></textarea>
					</label>

					<label>
						Additional References
						<textarea name="reference"></textarea>
					</label>

					<hr>
					<label>TLP Alert</label>
					<label>
						<input type="radio" value="amber" name="tlp" checked/>
						Amber
					</label>
					<label>
						<input type="radio" value="green" name="tlp"/>
						Green
					</label>
					<hr>
					
					<div class="grid">
						<input type="submit">
						<input type="reset">
					</div>
					
				</fieldset>
			</form>
		</article>
        `

    return Execute("credleak", page, data)
}
